pkg/server/qr_submission.go

```diff
@@ -0,0 +1,122 @@
+package server
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/Shopify/goose/srvutil"
+	"github.com/cds-snc/covid-alert-server/pkg/keyclaim"
+	"github.com/cds-snc/covid-alert-server/pkg/persistence"
+	pb "github.com/cds-snc/covid-alert-server/pkg/proto/covidshield"
+	"github.com/gorilla/mux"
+	"google.golang.org/protobuf/proto"
+)
+
+type qrSubmissionServlet struct {
+	db   persistence.Conn
+	auth keyclaim.Authenticator
+}
+
+func NewQRSubmissionServlet(db persistence.Conn, qrSubmissionAuth keyclaim.Authenticator) srvutil.Servlet {
+	s := &qrSubmissionServlet{db: db, auth: qrSubmissionAuth}
+
+	return srvutil.PrefixServlet(s, "/qr")
+}
+
+func (s *qrSubmissionServlet) RegisterRouting(r *mux.Router) {
+	r.HandleFunc("/new-submission", s.newExposureEvent)
+}
+
+func qrUploadResponse(errCode pb.QrSubmissionResponse_ErrorCode) *pb.QrSubmissionResponse {
+	return &pb.QrSubmissionResponse{Error: &errCode}
+}
+
+func (s *qrSubmissionServlet) newExposureEvent(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	if r.Method != "POST" {
+		log(ctx, nil).WithField("method", r.Method).Info("disallowed method")
+		http.Error(w, "unauthorized", http.StatusUnauthorized)
+		return
+	}
+
+	hdr := r.Header.Get("Authorization")
+	_, originator, ok := s.auth.RegionFromAuthHeader(hdr)
+	if !ok {
+		log(ctx, nil).WithField("header", hdr).Info("bad auth header")
+		http.Error(w, "unauthorized", http.StatusUnauthorized)
```

Comment:
You removed the logging of the header field from pkg/server/keyclaim.go but left it here, should this be removed as well.

Reply:
Added back in because we are just logging failed attempts